### PR TITLE
Add space after '--' in single-line comments

### PR DESCRIPTION
**Pull request by brandonchinn178**
Fixes https://github.com/fourmolu/fourmolu/issues/212

### CODE REVIEW

File: src/Ormolu/Diff/ParseResult.hs [hunk 1]
```diff
@@ -125,7 +126,7 @@ matchIgnoringSrcSpans a = genericQuery a
       case cast str1' :: Maybe HsDocString of
         Nothing -> Different []
         Just str1 ->
-          if splitDocString str0 == splitDocString str1
+          if (map T.strip . splitDocString) str0 == (map T.strip . splitDocString) str1
```
Comment (brandonchinn178) [inline src/Ormolu/Diff/ParseResult.hs:129]: We won't care about leading/trailing whitespace in comments in the idempotence test. We should see any unexpected whitespace in test output anyway. If we find idempotence issues later, we could change this to something like
```hs
let normalize s =
      case T.uncons s of
        Just (' ', s') -> s'
        _ -> s
 in if (map normalize . splitDocString) str0 == (map normalize . splitDocString) str1
```

File: src/Ormolu/Utils.hs [hunk 2]
```diff
@@ -85,6 +86,20 @@ splitDocString docStr =
...
+          let leadingSpace txt = case T.uncons txt of
+                Just (' ', _) -> True
+                _ -> False
+              dropSpace txt =
+                if leadingSpace txt
+                  then T.drop 1 txt
+                  else txt
+           in if leadingSpace x
+                then dropSpace <$> xs
+                else xs
```
Comment (brandonchinn178) [inline src/Ormolu/Utils.hs:102]: This is on the Ormolu side. Must've been dropped in a bad merge
https://github.com/tweag/ormolu/blob/54642a7966d1d0787ba84636a2eb52f6729f600e/src/Ormolu/Utils.hs#L89-L102

Comment (georgefst) [reply src/Ormolu/Utils.hs:104]: Probably me back when I first implemented multi-line-style haddock. Good find.

**Comment by georgefst** [comment]
Nice! This fixes something that has been minorly annoying me for two years.

**Review by georgefst — APPROVED** [review]

**Pull request merged**